### ISSUE TITLE
Inject via annotation

### ISSUE THREAD

**Issue by gzm55** [marcelmay/inject-maven-plugin#29]
Hi, can we inject via some annotation like `@InjectMavenProperty`:

```java
@InjectMavenProperty( name = 'project.scm.url', default = null )
public string VER;
```

**Comment by marcelmay** [comment]
Hi @gzm55!

Currently the injected value and location can only be configured by plugin configuration.

I see the advantage of being able to specify the property on the Java side (no mapping in plugin definition, easier to configure etc), and the disadvantage of having an additional compile dependency.

The plugin could support both options - by existing plugin config, and by new by annotation.

**Comment by gzm55** [comment]
yes, a compiler scope dependency is not a big trouble, and the user would select ether proper method on different develop environments.

**Comment by gzm55** [comment]
@marcelmay can we you `javax.inject.Named` to mark a plugin injected value/method?
